src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,125 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final Double MULTIPLIER = 1.5;
+    private Object[] internalArray;
+    private int size;
+
+    public ArrayList() {
+        this.internalArray = new Object[10];
+    }
+
     @Override
     public void add(T value) {
-
+        grow();
+        internalArray[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index doesn't exist, size of array = "
+                    + size);
+        }
+        grow();
+        if (size == index) {
+            internalArray[size] = value;
+        } else {
+            Object[] internalArrayTemp = new Object[internalArray.length];
+            System.arraycopy(internalArray,0, internalArrayTemp, 0, index);
+            System.arraycopy(internalArray,index, internalArrayTemp, index + 1, size - index);
+            internalArrayTemp[index] = value;
+            internalArray = internalArrayTemp;
+        }
+        size++;
```

Comment:
You can simplify it:
```suggestion
    System.arraycopy(internalArray, index, internalArray, index + 1, size - index);
    internalArray[index] = value;
    size++;
```